examples/image_viewer.rs

```diff
@@ -177,23 +175,14 @@ fn main() {
     // First, we initialize a few boolean flags that we'll use to track our state:
     // - the window needs to be redrawn
     let mut need_redraw = false;
-    // - the frame needs to be refreshed
-    let mut need_refresh = false;
-    // - have we done the original bootstrap? Depending on the underlying shell
-    //   protocol, we may need to wait until we receive a first configure event
-    //   before drawing our window. This flag checks if this is done.
-    let mut bootstrapped = false;
-    // - number of currently free memory pool, which will be shared by the buffer
-    //   implementations
-    let free_pools = Rc::new(Cell::new(2u32));
     // - are we currently in the process of being resized? (to draw the image or
     //   black content)
     let mut resizing = false;
-    // - a QueueToken, we'll need it to implement the buffers
-    let token = event_queue.get_token();
     // - the size of our contents
     let mut dimensions = image.dimensions();
 
+    let mut buffer = None;
```

Comment:
Given `MemPool` now destroys buffers by itself, we don't need to keep this variable any more.